app/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/meldekort/domene/MeldekortBehandling.kt

```diff
@@ -355,3 +361,40 @@ fun Sak.opprettMeldekortBehandling(
         ),
     )
 }
+
+fun Sak.opprettMeldekortKorrigering(
+    saksbehandler: Saksbehandler,
+    forrigeBehandling: MeldekortBehandling,
+): MeldekortBehandling.MeldekortUnderBehandling {
+    require(this.meldekortBehandlinger.contains(forrigeBehandling)) {
+        "Forrige behandling ${forrigeBehandling.id} tilhører ikke denne saken"
+    }
+
+    val meldekortId = MeldekortId.random()
+    val meldeperiode = hentSisteMeldeperiodeForKjede(forrigeBehandling.kjedeId)
+
+    return MeldekortBehandling.MeldekortUnderBehandling(
+        id = meldekortId,
+        sakId = this.id,
+        saksnummer = this.saksnummer,
+        fnr = this.fnr,
+        opprettet = nå(),
+        // TODO: bør navkontor hentes på nytt? Kan den være endret innenfor samme meldeperiode?
+        navkontor = forrigeBehandling.navkontor,
+        ikkeRettTilTiltakspengerTidspunkt = forrigeBehandling.ikkeRettTilTiltakspengerTidspunkt,
```

Comment:
Dette bør du bruke siste meldeperioden i meldeperiodekjeden for å utlede. Det er ikke sikkert forrige behandling peker på siste meldeperiode.